plugins/algoltek-usb/fu-algoltek-usb-device.c

```diff
@@ -91,11 +93,30 @@ fu_algoltek_usb_device_rdv(FuAlgoltekUsbDevice *self, GError **error)
 						  error))
 		return NULL;
 
-	/* Remove len and cmd bytes */
-	for (guint32 i = 2; i < st->len; i++) {
-		if (st->data[i] < 128)
-			fu_byte_array_append_uint8(version_data, st->data[i]);
+	if (!fu_memread_uint16_safe(st->data, st->len, 2, &version_prefix, G_BIG_ENDIAN, error))
+		return NULL;
+
+	if (version_prefix == 0x4147) {
+		/* remove len, cmd bytes and "AG" prefixes */
+		for (guint32 i = 4; i < st->len; i++) {
+			if (st->data[i] == '_') {
+				underscore_count++;
+				if (underscore_count == 1)
+					continue;
```

Comment:
I'm confused by this; we start at 0, increment to 1 -- and then immediately loop around again. I think you can remove the `if` statement here.

Reply:
If I'm not jump to the next loop,the first '_' would be add to the version text,too.